src/rotating_proxy.py

```diff
@@ -0,0 +1,106 @@
+# import random
+from typing import Callable
+import requests
+import threading
+import time
+import itertools
+
+import urllib.request
+import gateio_new_coins_announcements_bot.globals as globals
+from gateio_new_coins_announcements_bot.logger import logger
+
+
+_proxy_list = {}
+_proxy = None
+event = threading.Event()
```

Comment:
`event` should be marked private too and functions should be created to interact with it from the outside to make it easier to implement unit tests.